crates/primitives/snos-output/src/codec.rs

```diff
@@ -0,0 +1,181 @@
+use alloc::vec::Vec;
+
+use parity_scale_codec::{Decode, Encode, Input, Output};
```

Comment:
I have the feeling you are using the `parity_scale_codec` `Encode` and `Decode` traits, but the way you implement them is not using the [SCALE encoding](https://docs.substrate.io/reference/scale-codec/).

Am I right? If so, you should not use those traits, and instead create some trait or method of your own for this purpose.

Reply:
Yep, that was just the easiest way to both use types in runtime api and abi encode them.
I will derive scale trait then and add a custom serialization trait implementation